### PR TITLE
Implement PutCard method

### PR DESCRIPTION
**Pull request by matyimat**

### CODE REVIEW

File: Dixit_Logic/Classes/DixitGame.cs [hunk 1]
```diff
@@ -106,23 +106,24 @@ public bool RemovePlayer(IPlayer player)
...
-            }
-            else
+            if(!_actGameState.GameIsRuning)
             {
-                return false;
+                int playerCount = _actGameState.Players.Count;
+
+                if (playerCount >= _minPlayerNumber && playerCount <= _maxPlayerNumber)
+                {
+                    _actGameState.RoundStatus = PhaseStatus.AssociationTelling;// ha még nem indított már el valaki korábban
```
Comment (VSZM) [inline Dixit_Logic/Classes/DixitGame.cs]: How do we know who is exactly the story teller?

File: Dixit_Logic/Classes/DixitGame.cs [hunk 2]
```diff
@@ -134,16 +135,20 @@ public void EvaluatePoints()
...
         /// </summary>
         /// <param name="storyText">An association text string</param>
+        /// <param name="player">The player who guess the text</param>
         /// <returns>True if game's round status is in AssociationTelling phase and game
         /// is runing. Otherwise return false.  </returns>
-        public bool AddAssociationText(string storyText)
+        public bool AddAssociationText(string storyText, IPlayer player)
         {
-            if(_actGameState.GameIsRuning && _actGameState.RoundStatus == PhaseStatus.AssociationTelling)
+            if(_actGameState.RoundStatus == PhaseStatus.AssociationTelling)
```
Comment (VSZM) [inline Dixit_Logic/Classes/DixitGame.cs]: I think in this if clause we need to check if the player in the parameter is equal to the player who we are expecting to hear the story from.

File: Dixit_Logic/Classes/DixitGame.cs [hunk 3]
```diff
@@ -134,16 +135,20 @@ public void EvaluatePoints()
...
             {
                 _actGameState.CardAssociationText = storyText;
+                _actGameState.ActualPlayer = player;
```
Comment (VSZM) [inline Dixit_Logic/Classes/DixitGame.cs]: Actual player should already be known here.

File: Dixit_Logic/Classes/DixitGame.cs [hunk 4]
```diff
@@ -170,7 +175,27 @@ public void NewGuess(IPlayer player, ICard card)
...
+            if(_actGameState.RoundStatus == PhaseStatus.Putting)
+            {
+                IDeck playerHand;
+
+                if(player != null && _actGameState.Hands.TryGetValue(player, out playerHand))
+                {
+                    ICard selectedCard = ((Hand)playerHand).GetCard(card);
+                    if (selectedCard != null)
+                    {
+                        _actGameState.BoardDeck.AddCard(selectedCard);
```
Comment (VSZM) [inline Dixit_Logic/Classes/DixitGame.cs:191]: During point calculations at the end of round, how will we know the relation between the cards on the board, and the players that put them down?

File: Dixit_Logic/Classes/GameState.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 namespace Dixit_Logic.Classes
 {
-    public enum PhaseStatus { AssociationTelling = 0, Putting = 1, Guessing = 2, RoundEnd = 3 };
+    public enum PhaseStatus { BeforeStart = 0, AssociationTelling = 1, Putting = 2, Guessing = 3, RoundEnd = 4 };
```
Comment (VSZM) [inline Dixit_Logic/Classes/GameState.cs]: Don't we need a Phase here for signaling the game is over?

**Review by VSZM — CHANGES_REQUESTED** [review]

File: Dixit_Logic/Classes/DixitGame.cs (hunk 4 above)
Comment (matyimat) [reply Dixit_Logic/Classes/DixitGame.cs:191]: I will add a player information to the cards what will indicate the player who drawed them.

**Review by VSZM — APPROVED** [review]

**Pull request merged**